### PR TITLE
Allow overriding test images

### PR DESCRIPTION
**Pull request by kfox1111**
Some of the tests can't run on an isolated network. Make the images overridable.

### CODE REVIEW

**Comment by kfox1111** [comment]
Reworked https://github.com/spiffe/helm-charts/pull/155. It had some github issue that made us have to abandon it.

File: charts/spire/charts/tornjak-frontend/values.yaml [hunk 1]
```diff
@@ -71,3 +71,17 @@ apiServerURL: "http://localhost:10000/"  # 👈 Use it for minikube or kind
...
+      # -- The OCI registry to pull the image from
+      registry: cgr.dev
+      # -- The repository within the registry
+      repository: chainguard/bash
+      # -- The image pull policy
+      pullPolicy: IfNotPresent
+      # -- This value is deprecated in favor of tag. (Will be removed in a future release)
+      version: ""
+      # -- Overrides the image tag
+      tag: latest-20230517
```
Comment (marcofranssen) [inline charts/spire/charts/tornjak-frontend/values.yaml:87]:
```suggestion
      tag: 5.2.15
```

Found this tag via 

```bash
crane ls cgr.dev/chainguard/bash | grep -v latest | grep -v sbom | grep -v att | grep -v sig
```

File: charts/spire/charts/spire-server/values.yaml [hunk 2]
```diff
@@ -340,3 +340,17 @@ tornjak:
...
+      # -- The OCI registry to pull the image from
+      registry: cgr.dev
+      # -- The repository within the registry
+      repository: chainguard/bash
+      # -- The image pull policy
+      pullPolicy: IfNotPresent
+      # -- This value is deprecated in favor of tag. (Will be removed in a future release)
+      version: ""
+      # -- Overrides the image tag
+      tag: latest-20230517
```
Comment (marcofranssen) [inline charts/spire/charts/spire-server/values.yaml:356]:
```suggestion
      tag: 5.2.15
```

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml [hunk 3]
```diff
@@ -185,3 +185,17 @@ ingress:
...
+      # -- The OCI registry to pull the image from
+      registry: cgr.dev
+      # -- The repository within the registry
+      repository: chainguard/bash
+      # -- The image pull policy
+      pullPolicy: IfNotPresent
+      # -- This value is deprecated in favor of tag. (Will be removed in a future release)
+      version: ""
+      # -- Overrides the image tag
+      tag: latest-20230517
```
Comment (marcofranssen) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:201]:
```suggestion
      tag: 5.2.15
```

File: charts/spire/charts/spiffe-oidc-discovery-provider/README.md [hunk 4]
```diff
@@ -76,6 +76,11 @@ A Helm chart to install the SPIFFE OIDC discovery provider.
 | telemetry.prometheus.podMonitor.labels | object | `{}` |  |
 | telemetry.prometheus.podMonitor.namespace | string | `""` | Override where to install the podMonitor, if not set will use the same namespace as the spiffe-oidc-discovery-provider |
 | telemetry.prometheus.port | int | `9988` |  |
+| tests.bash.image.pullPolicy | string | `"IfNotPresent"` | The image pull policy |
+| tests.bash.image.registry | string | `"cgr.dev"` | The OCI registry to pull the image from |
+| tests.bash.image.repository | string | `"chainguard/bash"` | The repository within the registry |
+| tests.bash.image.tag | string | `"latest-20230517"` | Overrides the image tag |
```
Comment (marcofranssen) [inline charts/spire/charts/spiffe-oidc-discovery-provider/README.md:82]:
```suggestion
| tests.bash.image.tag | string | `"5.2.15"` | Overrides the image tag |
```

File: charts/spire/README.md [hunk 5]
```diff
@@ -214,6 +214,11 @@ Kubernetes: `>=1.21.0-0`
 | spiffe-oidc-discovery-provider.telemetry.prometheus.podMonitor.labels | object | `{}` |  |
 | spiffe-oidc-discovery-provider.telemetry.prometheus.podMonitor.namespace | string | `""` | Override where to install the podMonitor, if not set will use the same namespace as the spiffe-oidc-discovery-provider |
 | spiffe-oidc-discovery-provider.telemetry.prometheus.port | int | `9988` |  |
+| spiffe-oidc-discovery-provider.tests.bash.image.pullPolicy | string | `"IfNotPresent"` | The image pull policy |
+| spiffe-oidc-discovery-provider.tests.bash.image.registry | string | `"cgr.dev"` | The OCI registry to pull the image from |
+| spiffe-oidc-discovery-provider.tests.bash.image.repository | string | `"chainguard/bash"` | The repository within the registry |
+| spiffe-oidc-discovery-provider.tests.bash.image.tag | string | `"latest-20230517"` | Overrides the image tag |
```
Comment (marcofranssen) [inline charts/spire/README.md:220]:
```suggestion
| spiffe-oidc-discovery-provider.tests.bash.image.tag | string | `"5.2.15"` | Overrides the image tag |
```

File: charts/spire/README.md [hunk 6]
```diff
@@ -415,6 +425,11 @@ Kubernetes: `>=1.21.0-0`
 | tornjak-frontend.serviceAccount.create | bool | `true` | Specifies whether a service account should be created |
 | tornjak-frontend.serviceAccount.name | string | `""` | The name of the service account to use. If not set and create is true, a name is generated using the fullname template |
 | tornjak-frontend.spireHealthCheck.enabled | bool | `true` | Enables the SPIRE Healthchecker indicator |
+| tornjak-frontend.tests.bash.image.pullPolicy | string | `"IfNotPresent"` | The image pull policy |
+| tornjak-frontend.tests.bash.image.registry | string | `"cgr.dev"` | The OCI registry to pull the image from |
+| tornjak-frontend.tests.bash.image.repository | string | `"chainguard/bash"` | The repository within the registry |
+| tornjak-frontend.tests.bash.image.tag | string | `"latest-20230517"` | Overrides the image tag |
```
Comment (marcofranssen) [inline charts/spire/README.md:431]:
```suggestion
| tornjak-frontend.tests.bash.image.tag | string | `"5.2.15"` | Overrides the image tag |
```

**Review by marcofranssen — CHANGES_REQUESTED** [review]
Some inline suggestions to pin to the semver version of the image. Do we need a `global.tests.bash` to align with the pattern as well?

**Comment by kfox1111** [comment]
> Some inline suggestions to pin to the semver version of the image. Do we need a `global.tests.bash` to align with the pattern as well?

The other images don't have overrides at that level either. Just global.spire.image.registry to set them all at once to a mirror server, which should work here too.

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml [hunk 7]
```diff
@@ -185,3 +185,17 @@ ingress:
   #  - secretName: chart-example-tls
   #    hosts:
   #      - oidc-discovery.example.org
+
+tests:
+  bash:
+    image:
+      # -- The OCI registry to pull the image from
```
Comment (mrsabath) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:192]:
```suggestion
      # -- The OCI registry to pull the tests image from
```

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml [hunk 8]
```diff
@@ -185,3 +185,17 @@ ingress:
...
+      registry: cgr.dev
+      # -- The repository within the registry
+      repository: chainguard/bash
+      # -- The image pull policy
```
Comment (mrsabath) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:196]:
```suggestion
      # -- The tests image pull policy
```

**Review by mrsabath — APPROVED** [review]
LGTM!

**Review by marcofranssen — CHANGES_REQUESTED** [review]
Not sure if this makes it better. This is more then just a simple default we discussed yesterday. This is a lot of boilerplating and code to maintain. In that case I would prefer to just add it to the values.yaml.

**Comment by marcofranssen** [comment]
I suggest we move all back to the valuews.yaml and use the approach from this PR https://github.com/spiffe/helm-charts/pull/311 to exclude them from docs.

**Comment by kfox1111** [comment]
Rolled back.

**Review by marcofranssen — APPROVED** [review]
LGTM

**Pull request merged**